.github/workflows/cicd.yml

```diff
@@ -0,0 +1,96 @@
+name: CICD
+on: [push]
+jobs:
+  Validate:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone Repo
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+
+      - name: Set Up NodeJs
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14.17.4'
+          cache: npm
+
+      - name: Install Dependencies
+        run: npm ci
+
+      - name: Lint
+        run: npm run lint
+
+      - name: Test
+        run: npm test
+
+      - name: Build
+        run: npm run build
+
+      - name: Cache Deployment
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+        uses: actions/cache@v2
+        env:
+          cache-name: deployment-cache
+        with:
+          path: './**/*'
+          key: ${{ github.event.after }}
+
+  Deploy:
```

Comment:
Originally I had this as 2 separate workflows in 2 separate files (`ci.yml`, `cd.yml`), but the UX for seeing two jobs is better than the UX for seeing 2 workflows.